hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionInfo.java

```diff
@@ -69,7 +66,15 @@
  */
 @InterfaceAudience.Public
 public interface RegionInfo extends Comparable<RegionInfo> {
-  RegionInfo UNDEFINED = RegionInfoBuilder.newBuilder(TableName.valueOf("__UNDEFINED__")).build();
+  /**
+   * @deprecated since 2.3.2/3.0.0; to be removed in 4.0.0 with no replacement (for internal use).
+   */
+  @Deprecated
+  @InterfaceAudience.Private
+  // Not using RegionInfoBuilder intentionally to avoid a static loading deadlock: HBASE-24627
```

Comment:
I think this statement can be omitted since this is being handled separately in the parent task.

Reply:
Makes sense.